packages/components/src/components/header/header.component.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { classMap } from 'lit/directives/class-map.js';
+import { html } from 'lit/static-html.js';
+import { property } from 'lit/decorators.js';
+import type { CSSResultGroup } from 'lit';
+import SynergyElement from '../../internal/synergy-element.js';
+import { HasSlotController } from '../../internal/slot.js';
+import componentStyles from '../../styles/component.styles.js';
+import styles from './header.styles.js';
+
+/**
+ * @summary The <syn-header /> element provides a generic application header
+ * that can be used to add applications name, toolbar and primary navigation.
+ *
+ * @documentation https://synergy-design-system.github.io/?path=/docs/components-syn-header--docs
+ * @status stable
+ * @since 1.8.0
+ *
+ *
+ * @slot - The label for the header. Will automatically be hidden on mobile.
+ * @slot logo - The logo that should be displayed. Will fall back to the SICK logo if not applied.
+ * @slot option-menu - Used to add various application toolbar icons.
+ *                     Best used with `<syn-icon-button />` and `<syn-drop-down />`
+ * @slot top-navigation - Used to add an optional horizontal navigation
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart side-navigation-button - Can be used to apply styles to the side-navigation menu icon
+ * @csspart logo - The wrapper where the application logo resides in
+ * @csspart label - Wrapper of the application name label
+ * @csspart option-menu - Item that wraps the optional application menu
+ * @csspart top-navigation - Wrapper that holds the optional top navigation section
+ */
+export default class SynHeader extends SynergyElement {
+  static styles: CSSResultGroup = [
+    componentStyles,
+    styles,
+  ];
+
+  static dependencies = {
+  };
+
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'logo', 'label', 'option-menu', 'top-navigation');
+
+  /**
+   * The headers label. If you need to display HTML, use the `label` slot instead.
+   */
+  @property() label = '';
+
+  render() {
+    const hasTopNavigation = this.hasSlotController.test('top-navigation');
+
+    return html`
+      <header
+        class=${classMap({
+          header: true,
+
+          // Is the top navigation available?
+          'header--has-top-navigation': hasTopNavigation,
+        })}
+        part="header"
+      >
+        <!-- .primary-content-area -->
+        <div part="primary-content-area" class="primary-content-area">
+
+          <div part="logo" class="header__logo">
```

Comment:
Following BEM this is not really correct, as you added a class in between, is it?

Reply:
I opted for the name `header__content` (as generic as it gets...)